conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/BodySerDe.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.reflect.TypeToken;
+import io.undertow.server.HttpServerExchange;
+import java.io.IOException;
+import java.io.InputStream;
+
+/** Request and response Deserialization and Serialization functionality used by generated code. */
+public interface BodySerDe {
+
+    /** Creates a {@link Serializer} for the requested type. Serializer instances should be reused. */
+    <T> Serializer<T> serializer(TypeToken<T> type);
```

Comment:
I've just noticed that `TypeToken` comes from Guava and is actually annotated `@Beta`, which I think means it might change across guava versions... 

Do you think it's worth lifting the TypeCapture#capture() method into our own lib? Or shall we just run this as-is, accepting the small risk that we'd need to update our interfaces if guava decides to break us?

Reply:
@uschi2000 Curious of your thoughts on using guava TypeToken vs defining our own, I think this is a relic from your prototype/dialogue.
We use guava collections in generated code, but they're not `@Beta`, and not core api.

I think we might as well define our own to avoid awkward breaks in the future, I'll PR into this branch to try it out.